cobbler/tftpgen.py

```diff
@@ -774,10 +774,9 @@ def write_pxe_file(
         if boot_loaders is None or format not in boot_loaders:
             return None
 
-        (rval, settings) = utils.input_string_or_dict(self.settings.to_dict())
-        if rval:
-            for key in list(settings.keys()):
-                metadata[key] = settings[key]
+        settings = utils.input_string_or_dict(self.settings.to_dict())
+        for key in list(settings.keys()):
+            metadata[key] = settings[key]
```

Comment:
Copying the keys into a new list is not necessary for this operation. The same value-copying can be done with just
```python
for k in settings:
    metadata[k] = settings[k]
# or
for k in settings.keys():
    metadata[k] = settings[k]
```

Neither of those are needed since `metadata` is a `dict`:
```suggestion
        metadata.update(settings)
```